test/e2e/tests/008-categoryDetailsEditCategory.js

```diff
@@ -0,0 +1,122 @@
+import * as PromiseUtils from '@core/promiseUtils'
+import * as StringUtils from '@core/stringUtils'
+
+import { click, expectExists, expectToBe, toRightOf } from '../utils/api'
+
+import {
+  addCategoryLevel,
+  addCategoryItem,
+  updateCategoryLevelName,
+  waitForLoader,
+  clickSidebarBtnDesignerCategories,
+  clickCategoryButtonDone,
+  clickCategoryItem,
+  clickCategoryItemBtnClose,
+  expectCategoryItemsInLevel,
+  expectCategoryItemsInLevelEmpty,
+} from '../utils/ui'
+
+const levelNames = {
+  0: 'country',
+  1: 'region',
+  2: 'district',
+}
+const itemsPerLevel = 5
+const itemInsertTime = 5 * 1000 // 5 sec
+
+const _getItemCode = ({ index, codePrefix = '' }) => codePrefix + StringUtils.padStart(2, '0')(String(index + 1))
+
+const _createItems = ({ levelIndex, codePrefix = '' }) =>
+  new Array(itemsPerLevel).fill().map((_, index) => {
+    const code = _getItemCode({ index, codePrefix })
+    const label = StringUtils.capitalizeFirstLetter(`${levelNames[levelIndex]} ${code}`)
+    return { code, label }
+  })
+
+const _addChildrenItems = async ({ codePrefix, levelIndex }) => {
+  const itemsChildren = _createItems({
+    codePrefix,
+    levelIndex,
+  })
+  await PromiseUtils.each(itemsChildren, async (itemChild, itemIndex) =>
+    addCategoryItem({
+      levelIndex,
+      itemIndex,
+      code: itemChild.code,
+      label: itemChild.label,
+    })
+  )
+  await expectCategoryItemsInLevel({ levelIndex, numberOfItems: itemsPerLevel })
+}
+
+describe('Categories: edit existing category', () => {
+  test('CategoryList: navigate to categories', async () => {
+    await waitForLoader()
+    await clickSidebarBtnDesignerCategories()
+    // Expect to have an invalid category
+    await expectExists({ text: 'Invalid' })
```

Comment:
You must be sure that a specific category is invalid

Reply:
ok, right